### ISSUE TITLE
Set market filter on double click of empty slot

### ISSUE THREAD

**Issue by Frosty-nee** [pyfa-org/Pyfa#898]
As an alternative to searching through the market tree for the module you want to fit, allow the user to double click an empty slot, or right click > fit module a group of empty slots.

This will filter for only modules that fit in those slots, and move the user's cursor to the search box for adding search terms to further narrow the search.

**Comment by blitzmann** [comment]
> As an alternative to searching through the market tree for the module you want to fit

... people actually use the market tree? I usually just search for it in the search bar - much quicker, and then right click the module and show module group if I want to see the meta items.

As for the proposal... it's very interesting and might be possible without it being too hackish. Would the actual market tree filtered, or just the results shown at the bottom? How do you propose a user would clear this? The problem with this as it is is that the search is not coupled with the market tree - the search is simply a search in the database, not a search of the market tree (which is why sometimes when the market tree bugs out and doesn't display an item, the search will usually bring it up)

We could also do something that I've been wanting to do for a while: specific search criteria. Like what Google has. you can search `slot:high 200` and get back high slot modules with 200 in the name. But again, the search doesn't mess with the market tree at all, and I'm not sure it should.

**Comment by Ebag333** [comment]
In this use case, the idea is that you are limiting the market tree to specific criteria, so you can more quickly fill out a fit. If you have 8 low slots to fill, and each is a unique module (not unusual) then you either
A) Search 8 times
B) Browse through ALL the modules looking for low slots. 

The idea here is just to add filters for slot, just like we have for faction, ded, and officer.  The double click on an empty slot is just a quick way of setting that filter on an otherwise wasted command. 

People have been asking for this filter for a long time. Double clicking on an empty slot is just a clever way of setting it. 

TL;DR: Not a search, but a filter.

**Comment by blitzmann** [comment]
I understand the concept, but I want to get a better picture of how it would be implemented, because it's not straightforward. The problem with the market tree is that we don't look forward to determine if that particular group shows up. We basically start by getting the root groups, then when those are expanded we get groups under that etc until we don't have any more groups, but that group has items. To be able to filter the market tree would require us to switch from a top-down hierarchy to a bottom-up hierarchy, because now we need to consider attributes on the items themselves. And I believe evaluating those items will cause performance issues.

A search is different though, and I would assume be much easier to implement depending on how we want to do so.  

And then there is still the question on how to "reset".

Here's a beautiful MSPaint mockup of a possible option: filter icon that had pre-defined filters (low, high, rig).

![image](https://cloud.githubusercontent.com/assets/3904767/21107517/f4ff87bc-c05f-11e6-9edd-fa3719f3268d.png)

But we still run into the bottom-up problem, and I don't particularly like this option.

I dunno, open to more ideas. Not opposed to reworking the market tree as long as performance can be maintained. It would be cool to be able to search for a module and have the market tree automatically change to reflect it.
